nuxeo-mongodb-audit/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditEntryReader.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc
+ */
+package org.nuxeo.mongodb.audit;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.bson.Document;
+import org.nuxeo.ecm.platform.audit.api.ExtendedInfo;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.impl.ExtendedInfoImpl;
+import org.nuxeo.ecm.platform.audit.impl.LogEntryImpl;
+
+import com.mongodb.DBObject;
+
+/**
+ * Reader for MongoDB Audit.
+ *
+ * @since 9.1
+ */
+public class MongoDBAuditEntryReader {
+
+    public static LogEntry read(Document doc) {
+        LogEntryImpl entry = new LogEntryImpl();
+        for (String key : doc.keySet()) {
+            if ("category".equals(key)) {
+                entry.setCategory(doc.getString(key));
+            } else if ("principalName".equals(key)) {
+                entry.setPrincipalName(doc.getString(key));
+            } else if ("comment".equals(key)) {
+                entry.setComment(doc.getString(key));
+            } else if ("docLifeCycle".equals(key)) {
+                entry.setDocLifeCycle(doc.getString(key));
+            } else if ("docPath".equals(key)) {
+                entry.setDocPath(doc.getString(key));
+            } else if ("docType".equals(key)) {
+                entry.setDocType(doc.getString(key));
+            } else if ("docUUID".equals(key)) {
+                entry.setDocUUID(doc.getString(key));
+            } else if ("eventId".equals(key)) {
+                entry.setEventId(doc.getString(key));
+            } else if ("repositoryId".equals(key)) {
+                entry.setRepositoryId(doc.getString(key));
+            } else if ("_id".equals(key)) {
+                entry.setId(doc.getLong(key));
+            } else if ("eventDate".equals(key)) {
+                entry.setEventDate(doc.getDate(key));
+            } else if ("logDate".equals(key)) {
+                entry.setLogDate(doc.getDate(key));
+            } else if ("extended".equals(key)) {
+                entry.setExtendedInfos(readExtendedInfo(doc.get(key, Document.class)));
+            }
```

Comment:
Properties names should be constants as they are shared with MongoDBAuditEntryWriter